keps/sig-node/127-usernamespaces-support/README.md

```diff
@@ -0,0 +1,981 @@
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Container security consists of many different kernel features that work together
+to make containers secure. User namespaces isolate user and group IDs by
+allowing processes to run with different IDs in the container and in the host.
+Specially, a process running as privileged in a container can be
+unprivileged in the host. This makes it possible to give more capabilities to the containers and
+protects the host and other containers from malicious or compromised containers.
+
+This KEP adds a new `userNamespaceMode` field  to `pod.Spec`. It allows users to
+place pods in different user namespaces increasing the  pod-to-pod and
+pod-to-host isolation. This extra isolation increases the cluster security as it
+protects the host and other pods from malicious or compromised processes inside
+containers that are able to break into the host. This KEP proposes three
+different modes: `Host` keeps the current behaviour, `Cluster` uses the same
+ID mapping for all the pods (very similar to the previous [Support Node-Level User
+Namespaces
+Remapping](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/node-usernamespace-remapping.md)
+proposal) and `Pod` increases pod-to-pod isolation.
```

Comment:
We ccould say `Pod` attempts to give each pod a unique user namespace mapping.

Reply:
Right. I used  "different and non-overlapping" instead of "unique" in https://github.com/kinvolk/kubernetes-enhancements/pull/3/commits/c7cff4ff42e0ebe6079482f1547910472414bf01